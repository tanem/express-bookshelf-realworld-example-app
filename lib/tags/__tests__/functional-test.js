'use strict';

const chance = require('chance').Chance('tag');
const request = require('supertest');
const app = require('../../app');

const {tags} = app.locals.services;

describe('tags', () => {
  beforeEach(async () => {
    await tags.create({name: chance.word()});
    await tags.create({name: chance.word()});
  });

  describe('GET /api/tags', () => {
    test('should return a list of all tags', async () => {
      const {body, status} = await request(app)
        .get('/api/tags')
        .set('Accept', 'application/json')
        .send();

      expect({body, status}).toMatchSnapshot();
    });
  });
});
