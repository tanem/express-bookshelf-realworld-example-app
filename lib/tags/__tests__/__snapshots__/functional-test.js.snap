// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags GET /api/tags should return a list of all tags 1`] = `
{
  "body": {
    "tags": [
      "ehupezug",
      "jazas",
    ],
  },
  "status": 200,
}
`;
